src/AppInstallerCommonCore/Manifest/YamlParser.cpp

```diff
@@ -260,6 +253,28 @@ namespace AppInstaller::Manifest
             auto errors = ValidateAndProcessFields(installerNode, InstallerFieldInfos, fullValidation);
             std::move(errors.begin(), errors.end(), std::inserter(resultErrors, resultErrors.end()));
 
+            // Validate system reference strings if they are set at the installer level
+            if (fullValidation && !installer.PackageFamilyName.empty() && !ManifestInstaller::DoesInstallerTypeUsePackageFamilyName(installer.InstallerType))
+            {
+                resultErrors.emplace_back(ManifestError::InstallerTypeDoesNotSupportPackageFamilyName, "InstallerType", ManifestInstaller::InstallerTypeToString(installer.InstallerType));
+            }
+
+            if (fullValidation && !installer.ProductCode.empty() && !ManifestInstaller::DoesInstallerTypeUseProductCode(installer.InstallerType))
+            {
+                resultErrors.emplace_back(ManifestError::InstallerTypeDoesNotSupportProductCode, "InstallerType", ManifestInstaller::InstallerTypeToString(installer.InstallerType));
+            }

```

Comment:
In my previous change to separate manifest population vs manifest semantic validation, these checks fit in the ValidateManifest() in manifestvalidation.cpp

Reply:
Agreed, *but* this check can only be performed in this way here.  But I can go the stricter route and move them.

---
In reply to: [490695526](https://github.com/microsoft/winget-cli/pull/574#discussion_r490695526) [](ancestors = 490695526)

Reply:
Actually, I was wrong, and simply moving the validation parts will have the same net effect.

---
In reply to: [491164478](https://github.com/microsoft/winget-cli/pull/574#discussion_r491164478) [](ancestors = 491164478,490695526)